codex/sales/reservations.nim

```diff
@@ -248,7 +248,7 @@ proc exists*(
   let exists = await self.repo.metaDs.ds.contains(key)
   return exists
 
-iterator items(self: StorableIter): Future[?seq[byte]] =
+iterator items(self: StorableIter): auto =
```

Comment:
Out of curiosity, was there a need to change this to auto?

Reply:
A return type of `Future[?seq[byte]]` can raise a `CatchableException` when `await`ed. The return type of `self.next()` captures that it can only raise `CancelledError`, but it's not a simple `Future` type but generated by the `{.async.}` macro.